THREESCALE-11526 - Redis Sentinel with TLS for Porta and Backend

valerymo opened the pull request:
Jira: https://issues.redhat.com/browse/THREESCALE-11526

New document that  provides an overview of Redis Sentinel with TLS, detailing the installation and configuration process (without using a Helm Chart).

MStokluska commented on doc/redis-sentinel-tls.md:
```diff
@@ -0,0 +1,688 @@
+
+### Redis and Redis-Sentinel (Sentinel): Same Image, Different Configuration
+
+*We will use term **Sentinel** instead of full name **Sentinel** in most places in this doc.*
```
Redis Sentinel perhaps?

valerymo replied:
Thank you. Done.

MStokluska commented on doc/redis-sentinel-tls.md:
```diff
@@ -0,0 +1,688 @@
+If both the **Redis server** (redis) and **Sentinel** (redis-sentinel) are using **TLS** for communication, you'll need to configure both the Redis server and the Sentinel to use **SSL/TLS encryption** for secure connections. This involves setting up certificates and keys for both the Redis server and the Sentinel, and then ensuring the communication between them and the clients is secured via TL ...
+
+Here’s how to do this step-by-step in the OpenShift environment.
+The Guide includes not only Sentinel TLS configuration, but Redis server also, to be useful for Testing. 
```
```suggestion
The guide includes both Sentinel TLS configuration and Redis server, which are useful for testing. 
```

valerymo replied:
Thank you for suggestion. Yes, need change it. I  actually wanted to say before - that the Guide is for Sentinel. But there is No Sentinel without Redis server, so providing Redis-server configuration also. So after "consulting" with ChatGPT  I changed it to "The guide includes both Sentinel TLS configuration and Redis server setup, which is necessary for testing Sentinel. "   Hope it looks better now.

MStokluska commented on doc/redis-sentinel-tls.md:
```diff
@@ -0,0 +1,688 @@
+      containers:
+      - name: redis
+        #image: quay.io/fedora/redis-6
+        image: redis:7
```
lets move out of dockerhub - mirror.gcr.io/library/redis:7

MStokluska commented on doc/redis-sentinel-tls.md:
```diff
@@ -0,0 +1,688 @@
+    spec:
+      containers:
+      - name: redis
+        #image: quay.io/fedora/redis-6
```
can be removed?

valerymo replied:
Done. Thank you

MStokluska commented on doc/redis-sentinel-tls.md:
```diff
@@ -0,0 +1,688 @@
+    tls-auth-clients yes 
+    tls-replication yes
+
+    #sentinel monitor mymaster redis.3scale-test.svc.cluster.local 6380 2
```
might be worth to explain that this is using the DNS instead and is recommended. My ns is different so maybe something like "sentinel monitor mymaster redis.<NAMESPACE>.svc.cluster.local 6380 2"

MStokluska replied:
I've tried it with dns and it failed.

valerymo replied:
I tried more but didn't succed to work with DNS for master monitoring . 
New section added : 
![image](https://github.com/user-attachments/assets/dfe6fa8b-a0b8-49ed-9b3a-7cb38013e68b)

MStokluska commented on doc/redis-sentinel-tls.md:
```diff
@@ -0,0 +1,688 @@
+      containers:
+        - name: redis-sentinel
+          #image: quay.io/fedora/redis-6
+          image: redis:7
```
mirror.gcr.io/library/redis:7 and I think I would remove the above comment

valerymo replied:
done. Thank you

MStokluska commented on doc/redis-sentinel-tls.md:
```diff
@@ -0,0 +1,688 @@
+          #image: quay.io/fedora/redis-6
+          image: redis:7
+          ports:
+            - containerPort: 26379
```
In above cm we only have tls-port 26380 - do we need 26379 as well here?

valerymo replied:
non-TLS port can be useful for debugging in Testing environment. 
Thank you

MStokluska commented on doc/redis-sentinel-tls.md:
```diff
@@ -0,0 +1,688 @@
+- EC2 Ubuntu instance was created
+- Redis sources were downloaded, compiled, installed, run.
+- Sentinel's failover handling of redis was observed.  
+- We are providing only logs here. For installation details of Redis+Sentinel on EC2 - please see AWS and Redis docs or ChatGPT.
```
```suggestion
- We are providing only logs here. For installation details of Redis+Sentinel on EC2 - please see AWS and Redis docs.
```

valerymo replied:
updated. Thank you

MStokluska commented on doc/redis-sentinel-tls.md:
```diff
@@ -0,0 +1,688 @@
+openssl x509 -req -in redis-server.csr -CA ca.crt -CAkey ca.key -CAcreateserial -out redis-server.crt -days 365
+
+openssl genpkey -algorithm RSA -out redis-sentinel.key -pkeyopt rsa_keygen_bits:2048
+openssl req -new -key redis-sentinel.key -out redis-sentinel.csr -subj "/CN=redis-sentinel.3scale-test.svc.cluster.local"
```
^^ that comment makes no sense lol.
Basically, whenever we are mentioning DNS, maybe it would be better to expose DNS as part of your setup to run these commands and just use that instead? It's easy to miss it when doing for the first time

valerymo replied:
yes, meanwhile DNS is not working for master monitorying by Sentinel, so using IP.  "issues found" section added (see before). Thank you.

MStokluska reviewed (changes requested):
Really nice doc @valerymo - well done! 
Couple of minor points etc, the only thing that would be really nice to solve here is the DNS sentinels instead of IP, with DNS it did not work for me.

valerymo commented on doc/redis-sentinel-tls.md:
```diff
@@ -0,0 +1,688 @@
+### Prepare TLS Certificates
+Before you can configure Redis and Sentinel to use TLS, you'll need to create and configure certificates for the Redis server, Sentinel, and optionally for the client.
+
+**Note**: To correctly generate the Redis server and Sentinel certificates, you need to use their services IPs as the Common Names (CN) in the certificates. If you haven’t set up the services yet, please refer to **Section 4** to create the Redis server and Sentinel services, before continue to next item. Once the services are running, use their respective IPs as the CN when generating the certif ...
```
Updated. Thank you

valerymo commented on doc/redis-sentinel-tls.md:
```diff
@@ -0,0 +1,688 @@
+- `ca.crt`: The CA certificate
+- `redis-server.crt` and `redis-server.key`: The Redis server’s certificate and key
+- `redis-sentinel.crt` and `redis-sentinel.key`: The Sentinel’s certificate and key
+- `redis-client.crt` and `redis-client.key`: The Redis client's certificate and key
```
Updated/added:
+- Other (intermediate) files, such as *.csr and ca.srl, are also generated. If needed, additional information can be found in the OpenSSL documentation.
Thank you

valerymo commented:
@MStokluska  Thank you for the review! I believe I have addressed all comments, and the changes are made in a separate commit. Regarding the DNS issue with master monitoring by Sentinel, I tried more, but without success, so will continue using the IP for now. A new section titled 'Issues Found' has been added to the documentation, explaining the problem. 
Thanks
